assets/javascript/app.js

```diff
@@ -48,8 +91,86 @@ $("#authentication-btn").on("click", function() {
 
 //Initialize slide out menu
 $('#saved-recipes').sideNav({
-  menuWidth: 300, 
-  edge: 'left', 
-  closeOnClick: true 
+  menuWidth: 300,
+  edge: 'left',
+  closeOnClick: true
 }
-);
\ No newline at end of file
+);
+$(document).on("click",".save",function(e){
+name = $(e.target).data("name");
+userInput = $("#user-input").val().trim().toLowerCase();
+var searchURL = queryURLbase + userInput;
+console.log(userInput);
+saveRecipe(searchURL);
+
+console.log("name : "+name);
+
+
+})
+var name;
+function saveRecipe(queryURL){
+	$.ajax({
+			url: queryURL,
+			method: 'GET'
+	}).done(function(data) {
+			console.log(data);
+			console.log(queryURL);
+			imgAPI = data.hits[name].recipe.image;
+label = data.hits[name].recipe.label;
+recipe =data.hits[name].recipe.ingredients[0].text;
+sourceLink = data.hits[name].recipe.url;
+imgAPI = data.hits[name].recipe.image;
+var newRecipe = {
+	name:label,
+	ingredients: recipe,
+	link: sourceLink,
+	img: imgAPI
+}
+database.ref().push(newRecipe);
+console.log("label : "+label + " recipe : "+ recipe + " sourceLink : "+sourceLink);
+})
+}
+database.ref().on("child_added", function(childSnapshot) {
+	var name = childSnapshot.val().name;
+	var ingredients = childSnapshot.val().ingredients;
+	var link = childSnapshot.val().link;
+	var img = childSnapshot.val().img;
+	var key = childSnapshot.key;
+	console.log("link " + link);
+
+	var newList = $("<li>");
+	newList.attr("id",key);
+	var newSpan = $("<span>");
+	var linkA = $("<a>");
+	linkA.text(name);
+	linkA.attr("href",link);
+	var trash = $("<i>");
+	var pencil = $("<i>");
+	trash.attr("aria-hidden",true);
+	trash.addClass("fa fa-trash remove");
+	trash.attr("data-key",key);
+	pencil.attr("aria-hidden",true);
+	pencil.addClass("fa fa-pencil");
+	newList.append(newSpan);
+	newSpan.append(linkA,trash,pencil);
+	$("#recipeBox").append(newList);
+})
+
+$(document).on("click",".remove",function(e) {
```

Comment:
Because the trash can icon is right next to the recipe link in the recipe box section, I think it would be a good idea to have a confirmation modal here (just in case the user clicks on the trash can by mistake rather than clicking on the recipe link).